service/key/key.go

```diff
@@ -18,10 +18,26 @@ func ClusterID(customObject awstpr.CustomObject) string {
 	return customObject.Spec.Cluster.Cluster.ID
 }
 
+func MasterImageID(customObject awstpr.CustomObject) string {
+	return customObject.Spec.AWS.Masters[0].ImageID
```

Comment:
The key functions should probably return an empty string in case the master's slice is empty. Otherwise the function could panic.